library/build.go

```diff
@@ -202,6 +202,16 @@ func (b *Build) Environment(workspace, channel string) map[string]string {
 		envs["VELA_BUILD_TAG"] = tag
 	}
 
+	// check if the Build event is delete:tag
+	if strings.EqualFold(b.GetEvent(), constants.EventDelete) && strings.EqualFold(b.GetEventAction(), constants.ActionTag) {
+		// capture the tag reference
+		tag := b.GetBranch()
+
+		// add the tag reference to the list
+		envs["BUILD_TAG"] = tag
+		envs["VELA_BUILD_TAG"] = tag
```

Comment:
i think we should add more commentary explaining why we are doing a .GetBranch for populating build tag.